source/common/filter/http/filter_config_discovery_impl.cc

```diff
@@ -64,6 +78,21 @@ void DynamicFilterConfigProviderImpl::onConfigUpdate(Envoy::Http::FilterFactoryC
       });
 }
 
+void DynamicFilterConfigProviderImpl::onConfigRemoved(Config::ConfigAppliedCb cb) {
+  tls_.runOnAllThreads(
+      [cb, config = default_configuration_](OptRef<ThreadLocalConfig> tls) {
+        tls->config_ = config;
+        if (cb) {
+          cb();
+        }
+      },
+      [this]() {
+        // This happens after all workers have discarded the previous config so it can be safely
+        // deleted on the main thread by an update with the new config.
+        this->current_config_ = default_configuration_;
```

Comment:
Why is the new config the default config after TLS updates complete?

Reply:
The idea is to revert the state of the subscription to that state it would be in before it received an initial message or the fetch timeout is hit, which would mean reverting to the default configuration if one is provided. In the case that we don't have a default configuration `default_configuration_` should be `absl::nullopt` so this just clears out the config.

Reverting back to the default seems like the right thing to do: presumably people would be setting a "safe" configuration as the default, and would not want to revert to the "no config" state which would result in a "fail the request" filter to be used as a fallback.